tests/cases/project/realisticNpmTypes.json

```diff
@@ -0,0 +1,10 @@
+{
+    "scenario": "Realistic npm package types",
+    "projectRoot": "tests/cases/projects/realisticNpmTypes",
+    "inputFiles": [
+        "index.ts"
+    ],
+    "declaration": true,
+    "baselineCheck": true,
+    "moduleResolution": "node"
+}
```

Comment:
You are not specifying `allowJs`, that's why this is working even with the above bug, as you are never even checking for .js files.